packages/wonder-blocks-core/components/no-ssr.test.js

```diff
@@ -11,34 +11,60 @@ describe("NoSSR", () => {
         unmountAll();
     });
 
-    test("renders a placeholder first, then the actual content", (done) => {
+    test("renders a placeholder first, then the actual content", async () => {
         // Arrange
         const mockPlaceholder = jest.fn(() => null);
-
-        const arrangeAct = (assert) => {
+        await new Promise((resolve) => {
             const nodes = (
                 <NoSSR placeholder={mockPlaceholder}>
                     {() => {
-                        assert();
+                        resolve();
                         return null;
                     }}
                 </NoSSR>
             );
 
             // Act
             mount(nodes);
-        };
+        });
 
-        const assert = () => {
-            // Assert
-            // This was called from within the NoSSR children render prop.
-            // Therefore, if the placeholder has been called, it must have
-            // been called first.
-            expect(mockPlaceholder).toHaveBeenCalledTimes(1);
-            done();
-        };
+        // Assert
+        // Our promise doesn't resolve until the children render, therefore
+        // we don't get here until that and so if the placeholder has been
+        // called, it must have been called first.
+        expect(mockPlaceholder).toHaveBeenCalledTimes(1);
+    });
+
+    test("renders children right away if a parent NoSSR component handled first render", async () => {
+        // Arrange
+        const mockPlaceholder = jest.fn(() => null);
+        const mockPlaceholderNotCalled = jest.fn(() => null);
+        await new Promise((resolve) => {
+            const nodes = (
+                <NoSSR placeholder={mockPlaceholder}>
+                    {() => (
+                        <NoSSR placeholder={mockPlaceholderNotCalled}>
+                            {() => {
+                                resolve();
+                                return null;
+                            }}
+                        </NoSSR>
+                    )}
+                </NoSSR>
+            );
+
+            // Act
+            mount(nodes);
+        });
 
-        arrangeAct(assert);
+        // Assert
+        // Our promise doesn't resolve until the children of our nested NoSSR
+        // are rendered, therefore we don't get here until that and so if the
+        // parent placeholder has been called, it must have been called first.
+        // In addition, if our code is working right, the placeholder of the
+        // nested component should have been skipped.
+        expect(mockPlaceholder).toHaveBeenCalledTimes(1);
+        expect(mockPlaceholderNotCalled).not.toHaveBeenCalled();
```

Comment:
We should also check what's in `nodes` to verify that everything in the tree has been rendered.

Reply:
There isn't a snapshot test that checks this nested case.  Also, what's rendered in the snapshot is
```
This is rendered only by the client for all but the very first render of the component tree.
```
and what I'd actually like to check is what gets rendered after the first render.

Reply:
That is what gets rendered after the first render.

Reply:
@kevinbarabash Not sure what you're asking to be changed here, since that snapshot is already testing what gets rendered after the first render.

Reply:
Sorry for wording this imprecisely.  The snapshots handle the first render, I'm curious about what gets rendered on the second render.  Does `NoSSR` actually render its children like it's supposed to?

Reply:
I am confused. The snapshot is showing the subsequent renders, not the first.